File: app/src/test/java/fr/free/nrw/commons/LengthUtilsTest.java
package fr.free.nrw.commons;

import static org.hamcrest.CoreMatchers.is;

import fr.free.nrw.commons.location.LatLng;
import fr.free.nrw.commons.utils.LengthUtils;

import org.junit.Assert;
import org.junit.Test;

public class LengthUtilsTest {
    @Test public void testZeroDistance() {
        LatLng pointA = new LatLng(0, 0);
        LatLng pointB = new LatLng(0, 0);
        String distance = LengthUtils.formatDistanceBetween(pointA, pointB);
        Assert.assertThat(distance, is("0m"));
    }

    @Test public void testOneDegreeOnEquator() {
        LatLng pointA = new LatLng(0, 0);
        LatLng pointB = new LatLng(0, 1);
        String distance = LengthUtils.formatDistanceBetween(pointA, pointB);
        Assert.assertThat(distance, is("111.2km"));
    }

    @Test public void testOneDegreeFortyFiveDegrees() {
        LatLng pointA = new LatLng(45, 0);
        LatLng pointB = new LatLng(45, 1);
        String distance = LengthUtils.formatDistanceBetween(pointA, pointB);
        Assert.assertThat(distance, is("78.6km"));
    }

    @Test public void testOneDegreeSouthPole() {
        LatLng pointA = new LatLng(-90, 0);
        LatLng pointB = new LatLng(-90, 1);
        String distance = LengthUtils.formatDistanceBetween(pointA, pointB);
        Assert.assertThat(distance, is("0m"));
    }

    @Test public void testPoleToPole() {
        LatLng pointA = new LatLng(90, 0);
        LatLng pointB = new LatLng(-90, 0);
        String distance = LengthUtils.formatDistanceBetween(pointA, pointB);
        Assert.assertThat(distance, is("20,015.1km"));
    }
}

File: app/src/test/kotlin/fr/free/nrw/commons/LengthUtilsTest.kt
package fr.free.nrw.commons

import fr.free.nrw.commons.location.LatLng
import fr.free.nrw.commons.utils.LengthUtils
import org.junit.Assert.assertEquals
import org.junit.Test

class LengthUtilsTest {
    @Test
    fun testZeroDistance() {
        val pointA = LatLng(0.0, 0.0, 0f)
        val pointB = LatLng(0.0, 0.0, 0f)
        assertDistanceBetween("0m", pointA, pointB)
    }

    @Test
    fun testOneDegreeOnEquator() {
        val pointA = LatLng(0.0, 0.0, 0f)
        val pointB = LatLng(0.0, 1.0, 0f)
        assertDistanceBetween("111.2km", pointA, pointB)
    }

    @Test
    fun testOneDegreeFortyFiveDegrees() {
        val pointA = LatLng(45.0, 0.0, 0f)
        val pointB = LatLng(45.0, 1.0, 0f)
        assertDistanceBetween("78.6km", pointA, pointB)
    }

    @Test
    fun testOneDegreeSouthPole() {
        val pointA = LatLng(-90.0, 0.0, 0f)
        val pointB = LatLng(-90.0, 1.0, 0f)
        assertDistanceBetween("0m", pointA, pointB)
    }

    @Test
    fun testPoleToPole() {
        val pointA = LatLng(90.0, 0.0, 0f)
        val pointB = LatLng(-90.0, 0.0, 0f)
        assertDistanceBetween("20,015.1km", pointA, pointB)
    }

    private fun assertDistanceBetween(expected: String, pointA: LatLng, pointB: LatLng) =
            assertEquals(expected, LengthUtils.formatDistanceBetween(pointA, pointB))
}
